src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -435,6 +437,50 @@ function getLoopBuilder(state: TransformState, node: ts.Node, type: ts.Type): Lo
 	}
 }
 
+export function transformForOfRangeMacro(state: TransformState, node: ts.ForOfStatement): luau.List<luau.Statement> {
+	// we proved this before
+	assert(ts.isCallExpression(node.expression));
+
+	if (!ts.isVariableDeclarationList(node.initializer)) {
+		DiagnosticService.addDiagnostic(errors.noRangeMacroExpressionInitializer(node.initializer));
+		return luau.list.make();
+	}
+
+	// this should be impossible to fail
+	assert(node.initializer.declarations.length === 1);
```

Comment:
This can be done with @ts-ignore, I don't think a plain assert is the best way to go. It's easy enough to check into the above diagnostic.